riscv-configuration-structure-draft.adoc

```diff
@@ -268,9 +268,37 @@ big immediate benefit of a format that supports comments.
 [[sec:AccessMethod]]
 == Access Method
 
-The binary configuration structure is memory-mapped in system memory.
-There is a CSR which contains the physical address where the structure
-starts.
+The binary configuration structure is memory-mapped in system memory.  There is
+a CSR which contains the physical address where the structure starts.
+
+TODO: Get a CSR number assigned.
+
+When software running on a hart wants to read the configuration structure, it
+reads this CSR, and then decodes the binary structure at that physical address.
+The structure will specify which parts apply to which harts (identified by hart
+ID), and the software can ignore any information that does not apply to the hart
+it's running on.
+
+This provides hardware implementers several options for embedding the binary
+structure:
+1. The structure can describe all harts, and be accessible over the memory bus.
+All harts have the same memory map and the same CSR contents.
+2. There might be several structures in the system, and different harts are
+pointed to different structures by having different memory maps.
+3. There might be several structures in the system, and different harts are
+pointed to different structures by having different pointers in their CSR.
+4. A combination of 2 and 3 above could be used.
+
```

Comment:
In order to distinguish which method is provided by the processor package. We should declare a boolean in configuration-structure schema that indicates the CSs are aggregated to one structure and all the same if read from any harts. Or the CS is discrete in each hart. Because the hart 0 is mandatory, the boolean can only be provided by CS in hart0. With this FW knows how to read the complete CS, either from one hart, or run code in each hart to gather all CSs.

Reply:
That sounds reasonable, but how would this really be used? Like, either firmware can easily communicate with code running on other harts and this isn't really an issue, or it's impossible and then knowing that there are more harts doesn't help.

Reply:
FW can send the IPI to another hart to run the code to retrieve CS. That is doable and just depends on how do we implement it in FW.

Reply:
What does IPI mean in this context? How does the communication between harts occur? Whatever that communication mechanism is, can also be used to check if there are other harts.

Reply:
That refers to the interprocessor interrupt, which triggers the software interrupt on other harts. FW can use SBI to send the IPI to other harts for executing specific code. FW also knows how many harts are provided on the system because each hart runs fw code when it boots up.  So there are at least two solutions to read the discrete CS. How does FW read CS from hart 0 or other harts is implementation-specific and out of this TG scope.
Back to my original opinion, having a boolean indicates the CS is aggregated or discrete give FW an idea of the population of CS.  I will create the PR for the boolean if you agree with this, however, we don't have to mention this in the Access method paragraph.